sources/packages/api/src/route-controllers/student/student.controller.ts

```diff
@@ -134,4 +143,108 @@ export class StudentController extends BaseController {
   ): Promise<void> {
     await this.studentService.synchronizeFromUserInfo(userToken);
   }
+
+  @Get("/apply-PD-status")
+  async applyForPDStatus(@UserToken() userToken: IUserToken): Promise<void> {
+    // Get student details
+    const existingStudent = await this.studentService.getStudentByUserName(
+      userToken.userName,
+    );
+    if (!existingStudent) {
+      throw new NotFoundException(
+        `No student was found with the student name ${userToken.userName}`,
+      );
+    }
+    //check pd status in db, student should only allowed to check PD status once
+    // existingStudent?.StudentPDSentAt is set when student apply for PD Status first
+    // StudentPDStatus is null  before PD checker update status
+    // StudentPDStatus is true if PD Confirmed by ATBC
+    // StudentPDStatus is false if PD Denied by ATBC
+    // if student has a valid only, he/she should allow for a PD check
+    // if studentPDVerified is true from `SFASDB`, then no need to apply for PD to ATBC
+
+    if (
+      (existingStudent?.validSIN ?? false) &&
+      !existingStudent?.StudentPDSentAt &&
+      existingStudent?.StudentPDStatus === null &&
+      !existingStudent?.studentPDVerified
+    ) {
+      // create client payload
+      const payload: ATBCCreateClientPayload = {
+        SIN: existingStudent?.sin,
+        firstName: userToken?.givenNames,
```

Comment:
I believe that our database should be the source of truth for these data. Please consider getting these data from or database.